### ISSUE TITLE
listen for spacebar on trigger element

### ISSUE THREAD

**Issue by zebapy** [ghosh/Micromodal#193]
As mentioned in https://github.com/ghosh/Micromodal/issues/189 I'm using an anchor link for my modal trigger (for no-js fallback). Though this is usually not recommended because keyboard users must press `enter` to trigger the link, it seems appropriate to still use the link for the fallback.

Because of this we want the trigger element to function as a button, which needs to support spacebar press to trigger as well.

**Comment by gautamz07** [comment]
@zebapy Can you make a small demo describing your problem ? make a fiddle ?

**Comment by zebapy** [comment]
For elements that are modal triggers but not an actual button, I believe you should add `role=button` to them. Doing this means screen readers would insinuate that the user can press `spacebar` to trigger the modal as that's what `button` allows for. 

https://codepen.io/zebapy/pen/LoBdRy

Repro:
1. Use tab to focus the open modal link and press spacebar.

Expected:
- Modal should open

Result:
- Browser scrolls like normal (if there were more content)

**Comment by zebapy** [comment]
I opened a PR. I will note you could consider this an accessibility issue and should be added in case people use non-buttons for trigger elements.
